### PR TITLE
Add new `tate_text` dataset

### PR DESCRIPTION
**Pull request by juliasilge**
Addresses #10 

This PR *adds* a new `tate_text` dataset and *deprecates* both the `okc_text` and `okc` datasets. I don't believe there is any real official way to deprecate datasets, so I just put it in the documentation and we can be sure to socialize it when we do releases. The plan would then be to remove them entirely at the next release.

The new `tate_text` dataset really barely fits the way it is; I could only add the SVG we are using, not the other ones from lifecycle. I didn't import lifecycle but we do need it to run `devtools::document()`; do you want me to do that @topepo? Just for this one release cycle?

### CODE REVIEW

File: R/tate_text.R [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#' Tate Gallery modern artwork metadata
+#'
+#' Metadata such as artist, title, and year created for recent artworks owned
+#'  by the Tate Gallery. Only artworks created after 1980 are included, and the
+#'  metadata source was last updated in 2014. Tate does not endorse the
+#'  extracted data set here in this R package.
```
Comment (EmilHvitfeldt) [inline R/tate_text.R:6]: I'm a little confused about this sentence:

> Tate does not endorse the extracted data set here in this R package.

Does it mean that Tate doesn't endorse the data being in this package, or that Tate doesn't endorse the way it was extracted?

Comment (juliasilge) [reply R/tate_text.R:6]: I was trying to be extra careful about aligning with this:
https://github.com/tategallery/collection#usage-guidelines-for-open-data
Especially this part:
https://github.com/tategallery/collection#be-responsible

Any thoughts on how to be clear with this, or whether it is necessary in this context?

Comment (EmilHvitfeldt) [reply R/tate_text.R:6]: Maybe it is okay. All this legal stuff is hard to read anyways.

Maybe something along the lines of

> Tate doesn't endorse you or your use of the data set unless you have prior permission to do so.

Comment (topepo) [reply R/tate_text.R:6]: How about 

> The Tate Gallery provides these data but would like users to be respectful of their [guidelines for use](https://github.com/tategallery/collection#usage-guidelines-for-open-data).

**Review by topepo — APPROVED** [review]

**Pull request merged**